utils/pre-publish-check-ci.ts

```diff
@@ -8,33 +8,27 @@ import fg from "fast-glob";
 
 import {
     checkPrivate,
-    checkEntrypoints,
-    checkSource,
+    checkExports,
     checkPublishConfig,
 } from "./internal/pre-publish-utils";
 
-// eslint-disable-next-line promise/catch-or-return
-fg(path.join(__dirname, "..", "packages", "*", "package.json")).then(
-    (pkgPaths) => {
-        let allPassed = true;
-        // eslint-disable-next-line promise/always-return
-        for (const pkgPath of pkgPaths) {
-            // eslint-disable-next-line @typescript-eslint/no-require-imports
-            const pkgJson = require(path.relative(__dirname, pkgPath));
+const pkgPaths = fg.globSync(
+    path.join(__dirname, "..", "packages", "*", "package.json"),
+);
 
-            if (
-                !checkPrivate(pkgJson) &&
-                !checkPublishConfig(pkgJson) &&
-                !checkEntrypoints(pkgJson) &&
-                !checkSource(pkgJson)
-            ) {
-                allPassed = false;
-            }
-        }
+let allPassed = true;
+for (const pkgPath of pkgPaths) {
+    const pkgJson = require(path.relative(__dirname, pkgPath));
 
-        // Exit only after we've processed all the packages.
-        if (!allPassed) {
-            process.exit(1);
-        }
-    },
-);
+    // allPassed is at the end of the chain because of short-circuiting
```

Comment:
Why does `allPassed` have to be at the end of the chain? Do the `check*` functions have side effects that need to happen for each `package.json`? Note that some of the later `check`s might still not be called if the earlier ones return false.

Reply:
To make sure all the checks run, I'd suggest doing something like this:

```ts
function isFalsey(value: unknown): boolean {
  return !value
}

const results = pkgPaths.flatMap((pkgPath) => {
  const pkgJson = require(path.relative(__dirname, pkgPath));
  return [
    checkPrivate(pkgJson),
    checkPublishConfig(pkgJson),
    checkExports(pkgJson),
  ];
})

if (results.some(isFalsey)) {
  process.exit(1);
}
```

Reply:
I like this refactor!  This code was in place originally and when I started looking at it for this PR, I found that it wasn't even working (different callers had a different sense of what the `boolean` return value was - some interpreted `true` as a pass and some as a failure). 

I'll work in the suggestion you made as I think it makes things even clearer and avoids dealing with short-circuiting.